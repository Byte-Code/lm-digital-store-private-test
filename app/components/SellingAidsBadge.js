import React, { Component, PropTypes } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { Map } from 'immutable';
import glamorous from 'glamorous';

const Wrapper = glamorous.div({
  height: '179px',
  background: '#333333',
  display: 'flex',
  flexWrap: 'wrap',
  justifyContent: 'center',
  padding: '35px 0 47px'
});

const GroupName = glamorous.p({
  fontSize: '20px',
  fontFamily: 'LeroyMerlinSans Light',
  textAlign: 'center',
  lineHeight: '32px',
  color: '#fff',
  width: '100%'
});

const FilterWrapper = glamorous.div({
  display: 'flex',
  justifyContent: 'center',
  alignSelf: 'flex-end',
  '&>div': {
    marginRight: '10px'
  },
  '&>div:last-child': {
    marginRight: 0
  }
});

export const Filter = glamorous.div(({ isActive = false }) => ({
  minWidth: '150px',
  maxWidth: '300px',
  padding: '0 20px',
  height: '42px',
  borderRadius: '20px',
  backgroundColor: isActive ? '#67cb33' : '#efefef',
  color: isActive ? '#efefef' : '#333333',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  cursor: 'pointer',
  '&>p': {
    lineHeight: '20px',
    fontSize: '16px',
    textAlign: 'center'
  }
}));

export default class SellingAidsBadge extends Component {
  static propTypes = {
    sellingAids: ImmutablePropTypes.map,
    onToggle: PropTypes.func.isRequired,
    activeAid: PropTypes.string
  }

  static defaultProps = {
    sellingAids: Map(),
    activeAid: ''
  }

  renderAids() {
    const { sellingAids, onToggle, activeAid } = this.props;
    const aids = sellingAids.get('aids');

    return aids.map(aid => {
      const aidCode = aid.get('code');
      const isActive = aidCode === activeAid;

      return (
        <Filter
          key={aidCode}
          onClick={() => onToggle(aidCode)}
          isActive={isActive}
        >
          <p>{aid.get('name')}</p>
        </Filter>
      );
    });
  }

  render() {
    const { sellingAids } = this.props;
    if (sellingAids.isEmpty()) {
      return null;
    }
    const groupName = sellingAids.get('group');

    return (
      <Wrapper>
        <GroupName>{groupName}...</GroupName>
        <FilterWrapper>
          {this.renderAids()}
        </FilterWrapper>
      </Wrapper>
    );
  }
}
