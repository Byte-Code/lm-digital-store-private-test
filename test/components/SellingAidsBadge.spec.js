import React from 'react';
import { shallow } from 'enzyme';
import { Map, fromJS } from 'immutable';

import SellingAidsBadge, { Filter } from '../../app/components/SellingAidsBadge';

const emptySellingAids = Map();
const sellingAids = fromJS({
  aids: [
    { code: 'aid0', name: 'aid0', products: [0, 1, 2] },
    { code: 'aid1', name: 'aid1', products: [0, 4, 5] },
    { code: 'aid2', name: 'aid2', products: [0, 1, 9] }
  ]
});
const onToggle = jest.fn();
const activeAid = 'aid0';

describe('SellingAidsBadge', () => {
  it('should render null when sellingAids are empty', () => {
    const result = shallow(
      <SellingAidsBadge
        onToggle={onToggle}
        sellingAids={emptySellingAids}
        activeAid={activeAid}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should render a list of all the sellingAids otherwise', () => {
    const result = shallow(
      <SellingAidsBadge
        onToggle={onToggle}
        sellingAids={sellingAids}
        activeAid={activeAid}
      />
    );
    expect(result.find(Filter)).toHaveLength(3);
    expect(result).toMatchSnapshot();
  });

  it('should call onToggle once a filter is clicked', () => {
    const result = shallow(
      <SellingAidsBadge
        onToggle={onToggle}
        sellingAids={sellingAids}
        activeAid={activeAid}
      />
    );
    result.find(Filter).at(0).simulate('click');
    expect(onToggle).toHaveBeenCalledWith('aid0');
  });
});
