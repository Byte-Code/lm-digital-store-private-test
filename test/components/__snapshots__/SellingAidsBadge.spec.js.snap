// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellingAidsBadge should render a list of all the sellingAids otherwise 1`] = `
<styled.div>
  <styled.p>
    ...
  </styled.p>
  <styled.div>
    <styled.div
      isActive={true}
      onClick={[Function]}
    >
      <p>
        aid0
      </p>
    </styled.div>
    <styled.div
      isActive={false}
      onClick={[Function]}
    >
      <p>
        aid1
      </p>
    </styled.div>
    <styled.div
      isActive={false}
      onClick={[Function]}
    >
      <p>
        aid2
      </p>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`SellingAidsBadge should render null when sellingAids are empty 1`] = `null`;
